dev/com.ibm.ws.kernel.boot.core/src/com/ibm/ws/kernel/boot/internal/FileUtils.java

```diff
@@ -616,11 +616,12 @@ public Boolean run() throws Exception {
                 if (file.isDirectory()) {
                     success |= recursiveClean(file);
                 } else if ((file.getName().equals(BootstrapConstants.S_LOCK_FILE) &&
-                            fileToRemove.getName().equals(BootstrapConstants.LOC_AREA_NAME_WORKING))
+                            fileToRemove.getName().contains(BootstrapConstants.LOC_AREA_NAME_WORKING))
                            ||
-                           ((fileToRemove.getName().equals(BootstrapConstants.LOC_AREA_NAME_WORKING))) &&
+                           ((fileToRemove.getName().contains(BootstrapConstants.LOC_AREA_NAME_WORKING))) &&
                               (file.getName().equals(BootstrapConstants.SERVER_RUNNING_FILE))) {
-                    // skip/preserve workarea/.sLock and workarea/.sRunning files
+                    // skip/preserve workarea/.sLock and workarea/.sRunning files, including those
+                    // in embedded server workarea
```

Comment:
This logic is confusing.  Does the following if do the same (more precise thing)?

```
String candidate = file.getName();
String candidateParent = fileToRemove.getName();
if (
    (BootstrapConstants.S_LOCK_FILE.equals(candidate) || 
        BootstrapConstants.SERVER_RUNNING_FILE.equals(candidate))
    && 
    (BootstrapConstants.LOC_AREA_NAME_WORKING.equals(candidateParent) || 
        BootstrapConstants.LOC_AREA_NAME_WORKING_UTILS.equals(candidateParent))
  ) { ... }
```

Reply:
The intent was to minimize the complexity of the predicate; however, the solution above is more precise and probably faster.  Let's use it.